src/objc-renderer.ts

```diff
@@ -147,45 +151,48 @@ function toKeywordArgumentModifierString(argumentModifier:ObjC.KeywordArgumentMo
-  const typePart:string = renderableTypeReference(argument.type.reference);
+  const typePart:string = renderableTypeReference(genericizedType(generics, argument.type));
   const innerPart:string = modifiers.length > 0 ? modifiers + ' ' + typePart : typePart;
   return ':(' + innerPart + ')' + argument.name;
 }
 
-function toKeywordString(keyword:ObjC.Keyword):string {
-  return keyword.name + Maybe.match(toKeywordArgumentString, emptyString, keyword.argument);
+function toKeywordString(generics:string[], keyword:ObjC.Keyword):string {
+  return keyword.name + Maybe.match(FunctionUtils.pApplyf2(generics, toKeywordArgumentString), emptyString, keyword.argument);
 }
 
-function toClassMethodHeaderString(method:ObjC.Method):string {
+function toMethodHeaderString(modifier:string, method:ObjC.Method):string {
   const methodComments = method.comments.map(toCommentString).join('\n');
   const methodCommentsSection = codeSectionForCodeStringWithoutExtraSpace(methodComments);
 
-  return methodCommentsSection + '+ (' + toTypeString(method.returnType) + ')' + method.keywords.map(toKeywordString).join(' ') + ';';
+  return methodCommentsSection + modifier + ' (' + toTypeString(method.returnType) + ')' + method.keywords.map(FunctionUtils.pApplyf2([],toKeywordString)).join(' ') + ';';
 }
 
-function toInstanceMethodHeaderString(method:ObjC.Method):string {
-  const methodComments = method.comments.map(toCommentString).join('\n');
-  const methodCommentsSection = codeSectionForCodeStringWithoutExtraSpace(methodComments);
-
-  return methodCommentsSection + '- (' + toTypeString(method.returnType) + ')' + method.keywords.map(toKeywordString).join(' ') + ';';
-}
+var toClassMethodHeaderString    = FunctionUtils.pApplyf2('+', toMethodHeaderString);
+var toInstanceMethodHeaderString = FunctionUtils.pApplyf2('-', toMethodHeaderString);
 
-function toClassMethodImplementationString(method:ObjC.Method):string {
-  const methodStr =  '+ (' + toTypeString(method.returnType) + ')' + method.keywords.map(toKeywordString).join(' ') + '\n' +
+function toMethodImplementationString(modifier:string, generics:string[], method:ObjC.Method):string {
+  const methodStr =  modifier + ' (' + toGenericizedTypeString(generics, method.returnType) + ')' + method.keywords.map(FunctionUtils.pApplyf2(generics,toKeywordString)).join(' ') + '\n' +
                    '{\n' +
                    method.code.map(StringUtils.indent(2)).join('\n') +
                    '\n}';
   return methodStr;
 }
 
-function toInstanceMethodImplementationString(method:ObjC.Method):string {
-  const methodStr =  '- (' + toTypeString(method.returnType) + ')' + method.keywords.map(toKeywordString).join(' ') + '\n' +
-                   '{\n' +
-                   method.code.map(StringUtils.indent(2)).join('\n') +
-                   '\n}';
-  return methodStr;
+var toClassMethodImplementationString    = FunctionUtils.pApplyf3('+', toMethodImplementationString);
```

Comment:
Nit: we're generally not aligning consecutive assignments like this in the codebase